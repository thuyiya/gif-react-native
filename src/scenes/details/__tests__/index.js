/* eslint-disable no-undef */
import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import { shallow } from "enzyme";
import DetailScreen from "_scenes/details";
import { Giphy } from "_models";
import { VIEW_CONST } from "_constants";

const createTestProps = (props) => ({
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
  },
  route: {
    params: {
      item: new Giphy({}),
    },
  },
  ...props,
});

describe("SCREEN: Details", () => {
  const props = createTestProps({});

  it("enders correctly react-test-renderer", () => {
    const wrapper = new ShallowRenderer(<DetailScreen {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Child Element Getting Rendered Correctly", () => {
    const wrapper = shallow(<DetailScreen {...props} />);

    expect(
      wrapper.find({ testID: VIEW_CONST.DETAILS.giftCardTestID }).length
    ).toBe(1);
  });
});
