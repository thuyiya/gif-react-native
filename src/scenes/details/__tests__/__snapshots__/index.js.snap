// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SCREEN: Details enders correctly react-test-renderer 1`] = `
ReactShallowRenderer {
  "_context": null,
  "_didScheduleRenderPhaseUpdate": false,
  "_dispatcher": Object {
    "readContext": [Function],
    "useCallback": [Function],
    "useContext": [Function],
    "useDebugValue": [Function],
    "useDeferredValue": [Function],
    "useEffect": [Function],
    "useId": [Function],
    "useImperativeHandle": [Function],
    "useInsertionEffect": [Function],
    "useLayoutEffect": [Function],
    "useMemo": [Function],
    "useReducer": [Function],
    "useRef": [Function],
    "useResponder": [Function],
    "useState": [Function],
    "useSyncExternalStore": [Function],
    "useTransition": [Function],
  },
  "_element": null,
  "_firstWorkInProgressHook": null,
  "_forcedUpdate": false,
  "_idCounter": 0,
  "_instance": null,
  "_isReRender": false,
  "_newState": null,
  "_numberOfReRenders": 0,
  "_renderPhaseUpdates": null,
  "_rendered": null,
  "_rendering": false,
  "_updater": Updater {
    "_callbacks": Array [],
    "_renderer": [Circular],
  },
  "_workInProgressHook": null,
}
`;
